jth-tests/src/test/java/com/redhat/jenkins/nodesharing/GridTest.java

```diff
@@ -138,20 +151,168 @@ public void restartOrchestrator() throws Exception {
+        });
+        await(10000,
+                () -> queuedBlocker.isRunning(),
+                throwable -> { dumpFixtureLogs(); return "Build not running in time";
+        });
 
+        // Signal to finish run nr. 2
         queuedBlocker.complete();
+
+        // Run nr. 2 should complete with success as well
         await(20000, () -> buildDetails(executorClient.getJob("running"), 2).getResult() == BuildResult.SUCCESS, throwable -> { dumpFixtureLogs(); return "Build not completed in time"; });
     }
 
+    @Test(timeout = TEST_TIMEOUT * 2)
+    @ExternalFixture(name = "e0", roles = Executor.class,     resource = "executor-restartOrchestrator.yaml", injectPlugins = "matrix-auth")
+    @ExternalFixture(name = "o1", roles = Orchestrator.class, resource = "orchestrator.yaml",                 injectPlugins = "matrix-auth")
+    @ExternalFixture(name = "o2", roles = Orchestrator.class, resource = "orchestrator.yaml",                 injectPlugins = "matrix-auth")
+    public void changeOrchestratorUrlSmokeTest() throws Exception {
+        ExternalJenkinsRule.Fixture e0 = jcr.fixture("e0");
+        JenkinsServer executorClient0 = e0.getClient("admin", "admin");
+
+        ExternalJenkinsRule.Fixture o1 = jcr.fixture("o1");
+        JenkinsServer orchestratorClient1 = o1.getClient("admin", "admin");
+
+        ExternalJenkinsRule.Fixture o2 = jcr.fixture("o2");
+        JenkinsServer orchestratorClient2 = o1.getClient("admin", "admin");
+
+        JobWithDetails job = executorClient0.getJob("running");
+
+        // Run nr. 1 - will be in building state during Orchestrator restart
+        FileBuildBlocker runningBlocker = new FileBuildBlocker(tmp);
+        BuildWithDetails running = triggerJobAndWaitUntilStarted(executorClient0, "running",
+                job.build(runningBlocker.buildParams()));
+        await(10000,
+                () -> runningBlocker.isRunning(),
+                throwable -> { dumpFixtureLogs(); return "Build not running in time";
+        });
+
+        // Run nr. 2 - will be in queued state during Orchestrator restart
+        FileBuildBlocker queuedBlocker = new FileBuildBlocker(tmp);
+        QueueReference qr = job.build(queuedBlocker.buildParams());
+        executorClient0.getQueueItem(qr);
+
+        job = executorClient0.getJob("running");
+        // Check whether there are two runs for 'running' job:
+        //  - nr. 1. is building already
+        //  - nr. 2 sits in the queue before Orchestrator restart
+        assertTrue(running.isBuilding());
+        assertTrue(job.isInQueue());
+
+        job = executorClient0.getJob("running");
+
+        // The status for runs of 'running' job should be still the same and has to survive the Orchestrator restart
+        assertTrue(job.isInQueue());
+        assertTrue(buildDetails(job, 1).isBuilding());
+
+        assertThat(jcr.configRepo().getWorkTree().child("config").readToString(), containsString("orchestrator.url=" + o2.getUri()));
```

Comment:
`jcr.configRepo().getWorkTree().child("config")` should be extracted to variable for easier readability.